iceoryx_hoofs/posix/ipc/include/iox/detail/unix_domain_socket.inl

```diff
@@ -0,0 +1,160 @@
+        return err(PosixIpcChannelError::INTERNAL_LOGIC_ERROR);
+    }
+
+    auto tv = timeout.timeval();
+    auto setsockoptCall = IOX_POSIX_CALL(iox_setsockopt)(m_sockfd, SOL_SOCKET, SO_SNDTIMEO, &tv, sizeof(tv))
+                              .failureReturnValue(ERROR_CODE)
+                              .ignoreErrnos(EWOULDBLOCK)
+                              .evaluate();
+
+    if (setsockoptCall.has_error())
+    {
+        return err(errnoToEnum(setsockoptCall.error().errnum));
+    }
+    uint64_t msgSizeToSend = msgSize;
+    if (terminator == Termination::NULL_TERMINATOR)
+    {
+        msgSizeToSend += NULL_TERMINATOR_SIZE;
+    }
+    auto sendCall = IOX_POSIX_CALL(iox_sendto)(m_sockfd, msg, msgSizeToSend, 0, nullptr, 0)
+                        .failureReturnValue(ERROR_CODE)
+                        .evaluate();
+
+    if (sendCall.has_error())
+    {
+        return err(errnoToEnum(sendCall.error().errnum));
+    }
+    return ok();
+}
+
+template <typename Type>
+expected<uint64_t, PosixIpcChannelError> UnixDomainSocket::receiveImpl(Type* msg,
+                                                                       uint64_t maxMsgSize,
+                                                                       const units::Duration& timeout,
+                                                                       const Termination terminator) const noexcept
+{
+    if (PosixIpcChannelSide::CLIENT == m_channelSide)
+    {
+        IOX_LOG(ERROR, "receiving on client side not supported for unix domain socket \"" << m_name << "\"");
+        return err(PosixIpcChannelError::INTERNAL_LOGIC_ERROR);
+    }
+
+    auto tv = timeout.timeval();
+    auto setsockoptCall = IOX_POSIX_CALL(iox_setsockopt)(m_sockfd, SOL_SOCKET, SO_RCVTIMEO, &tv, sizeof(tv))
+                              .failureReturnValue(ERROR_CODE)
+                              .ignoreErrnos(EWOULDBLOCK)
+                              .evaluate();
+
+    if (setsockoptCall.has_error())
+    {
+        return err(errnoToEnum(setsockoptCall.error().errnum));
+    }
+
+    auto recvCall = IOX_POSIX_CALL(iox_recvfrom)(m_sockfd, msg, maxMsgSize, 0, nullptr, nullptr)
+                        .failureReturnValue(ERROR_CODE)
+                        .suppressErrorMessagesForErrnos(EAGAIN, EWOULDBLOCK)
+                        .evaluate();
+    if (recvCall.has_error())
+    {
+        if (terminator == Termination::NULL_TERMINATOR)
+        {
+            // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+            msg[0] = 0;
+        }
+        return err(errnoToEnum(recvCall.error().errnum));
+    }
+    if (terminator == Termination::NULL_TERMINATOR)
+    {
+        // NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-pointer-arithmetic)
+        if (msg[static_cast<uint64_t>(recvCall->value)] != 0)
+        {
+            return err(PosixIpcChannelError::INTERNAL_LOGIC_ERROR);
```

Comment:
> A few lines further above you add a null-termination in case of an error. Can you share your thoughts why you decided against it in this case?

Forget about it. I totally forgot that I proposed to not write a null-termination at the end but return an error.

The problem I overlooked is just that we might break an invariant of the `iox::string` since some data might be written to the buffer. The size would still be zero but the value at index 0 would not be the null-termination. Can you please write a 0 to `str[0]` right before `return 0;` in `timedReceive(iox::string<N>& ...`.